models/oneinch/ethereum/clipper/oneinch_clipper_ethereum_trades.sql

```diff
@@ -0,0 +1,281 @@
+    WHERE
+        call_success
+        {% if is_incremental() %}
+        AND call_block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        AND call_block_time >= '{{project_start_date}}'
+        {% endif %}
+            
+    UNION ALL
+        
+    SELECT
+        call_block_number,
+        output_returnAmount,
+        inputAmount as amount,
+        srcToken,
+        dstToken,
+        call_tx_hash,
+        call_trace_address,
+        call_block_time,
+        contract_address
+    FROM
+        {{ source('oneinch_ethereum', 'AggregationRouterV5_call_clipperSwapToWithPermit') }}
+    WHERE
+        call_success
+        {% if is_incremental() %}
+        AND call_block_time >= date_trunc("day", now() - interval '1 week')
+        {% else %}
+        AND call_block_time >= '{{project_start_date}}'
+        {% endif %}
+)
+, oneinch AS
+(
+    SELECT
+        call_block_number as block_number,
+        call_block_time as block_time,
+        '1inch' AS project,
+        'Clipper v1' AS version,
+        CAST(NULL as string) as taker,
+        CAST(NULL as string) as maker,
+        output_returnAmount AS token_bought_amount_raw,
+        amount AS token_sold_amount_raw,
+        CAST(NULL as double) AS amount_usd,
+        CASE
+            WHEN dstToken = '0x0000000000000000000000000000000000000000'
+            THEN '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+            ELSE dstToken
+        END AS token_bought_address,
+        CASE
+            WHEN srcToken = '0x0000000000000000000000000000000000000000'
+            THEN '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+            ELSE srcToken
+        END AS token_sold_address,
+        contract_address AS project_contract_address,
+        call_tx_hash as tx_hash,
+        call_trace_address AS trace_address,
+        CAST(-1 as integer) AS evt_index
+    FROM
+        clipper
+)
+
+SELECT
+    'ethereum' AS blockchain
+    ,src.project
+    ,src.version
+    ,date_trunc('day', src.block_time) AS block_date
+    ,src.block_time
+    ,src.block_number
+    ,token_bought.symbol AS token_bought_symbol
+    ,token_sold.symbol AS token_sold_symbol
+    ,case
+        when lower(token_bought.symbol) > lower(token_sold.symbol) then concat(token_sold.symbol, '-', token_bought.symbol)
+        else concat(token_bought.symbol, '-', token_sold.symbol)
+    end as token_pair
+    ,src.token_bought_amount_raw / power(10, token_bought.decimals) AS token_bought_amount
+    ,src.token_sold_amount_raw / power(10, token_sold.decimals) AS token_sold_amount
+    ,CAST(src.token_bought_amount_raw AS DECIMAL(38,0)) AS token_bought_amount_raw
+    ,CAST(src.token_sold_amount_raw AS DECIMAL(38,0)) AS token_sold_amount_raw
+    ,coalesce(
+        src.amount_usd
+        , (src.token_bought_amount_raw / power(10,
+            CASE
+                WHEN token_bought_address = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
```

Comment:
nit: should this be a variable? I can imagine a single e missing and this being hard to debug.

Reply:
agreed, i can do a passthrough of each script and apply some more variables. will also be good to set the practice for others to follow.